test-js/unit/pentaho/visual/base/View.Spec.js

```diff
@@ -0,0 +1,57 @@
+define([
+  "pentaho/visual/base/View"
+], function(View) {
+  "use strict";
+
+  describe("pentaho/visual/base/View", function() {
+    var element, dataSpec;
+
+    beforeEach(function() {
+      element = document.createElement("div");
+      var data = {};
+      dataSpec = {v: data};
+    });
+
+    describe("the constructor ", function() {
+
+      it("should throw if not invoked with two arguments", function() {
+        expect(function() {
+          return new View();
+        }).toThrowError(/required/);
+
+        expect(function() {
+          return new View(element);
+        }).toThrowError(/required/);
+
+        expect(function() {
+          return new View(element, dataSpec);
+        }).not.toThrow();
+      });
+
+      it("should throw if the first argument is not a DOM element", function() {
+        [
+          null, 1, true, {}, []
+        ].forEach(function(elem) {
+
+          expect(function() {
+            return new View(elem, dataSpec);
+          }).toThrowError(/Invalid/);
+        });
+      });
+
+    });
+
+    it("should not be renderable (returns a rejected promise when render() is called)", function() {
+      var view = new View(element, dataSpec);
```

Comment:
Here we go. Why is this being rejected? Nor the test title nor its assertion say why. I suspect it is throwing because dataSpec does not have a `dataSpec.meta.validate method`? And not because it is invalid, as I would expect :-)